modyn/supervisor/internal/triggers/ensembletrigger.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Generator
+from typing import cast
+
+from modyn.config.schema.pipeline.trigger.drift.config import DataDriftTriggerConfig
+from modyn.config.schema.pipeline.trigger.ensemble import EnsembleTriggerConfig
+from modyn.config.schema.pipeline.trigger.simple.data_amount import (
+    DataAmountTriggerConfig,
+)
+from modyn.config.schema.pipeline.trigger.simple.time import TimeTriggerConfig
+from modyn.supervisor.internal.triggers.amounttrigger import DataAmountTrigger
+from modyn.supervisor.internal.triggers.datadrifttrigger import DataDriftTrigger
+from modyn.supervisor.internal.triggers.timetrigger import TimeTrigger
+from modyn.supervisor.internal.triggers.trigger import Trigger, TriggerContext
+from modyn.supervisor.internal.triggers.utils.models import (
+    EnsembleTriggerEvalLog,
+    TriggerPolicyEvaluationLog,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class EnsembleTrigger(Trigger):
+    """Evaluates multiple subtriggers and triggers if a certain after an
+    aggregation function aggregates subtrigger decisions to True.
+
+    Idea:
+        The EnsembleTrigger always evaluates and caches the next triggering index
+        from the generators of each subtrigger. The evaluations of all subsequent subtrigger indexes are
+        deferred in a lazy manner.
+
+        We assume that after a subtrigger will always like to see a trigger after the next index in the generator.
```

Comment:
I don't think this sentence is grammatically correct? I at least don't understand it :D